### PR TITLE
verified domain ownership for full automatic networking

### PR DESCRIPTION
**Pull request by birarda**
- fixed a security issue where heartbeats sent to the ice-server from domains were not verified
- added failover for keypair generation from domain-server and Interface so new keypairs are generated if domain-server/ice-server fail to verify signature enough times
- moved account information to a separate file so account information is not stored with other settings

### CODE REVIEW

**Comment by birarda** [comment]
This will be complicated to test without the ice-server changes that are a part of this PR. I have tested the following locally:

- connecting to named domains and temporary domains
- denial of heartbeats from unverified domains
- connection to domain-server with restricted access
- re-generation of keypair when previous upload has failed or verification is failing

**Comment by hifi-gustavo** [comment]
Build console output for commit: f8392e62a86c55934fe83600e3cad6f15f6f0c24 available at: https://gist.github.com/3dac837d18d431058491<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7179.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7179.dmg

**Comment by birarda** [comment]
This is reviewable and testable (by connecting to my ice-server) but cannot be merged until Leo has confirmed that all HF production domains have an access token ready for their keypair generation.

File: domain-server/src/DomainServer.cpp [hunk 1]
```diff
@@ -110,7 +110,7 @@ DomainServer::DomainServer(int argc, char* argv[]) :
 
         // preload some user public keys so they can connect on first request
         _gatekeeper.preloadAllowedUserPublicKeys();
-
```
Comment (Atlante45) [inline domain-server/src/DomainServer.cpp]: New standard says no whitespace lines.
There's an option in Xcode to turn that off.

**Comment by hifi-gustavo** [comment]
Build console output for commit: f1aca57543894e648142e91c5c99855b68d1f052 available at: https://gist.github.com/3dac837d18d431058491<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7179.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7179.dmg

**Comment by Atlante45** [comment]
Code looks good, will do some testing tomorrow.

**Pull request merged**